fix(platform-browser): use correct parent in animation removeChild ca…

roman-sawicki opened the pull request:
…llback

Animation's `onRemovalComplete` callback is using incorrect `parentNode`.
Should be parentNode provided by `delegate.parentNode(element)` instead of direct `element.parentNode`.
This is not a problem with default renderer but can cause problems with custom renderer if it uses another
logic to add/remove nodes.

Fixes #44023

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?

Issue Number: 44023


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

crisbeto commented:
The change looks good, but the CI flaked. @raman-savitski can you try to push the same commit again so the CI can run?

romaneos committed:
fix(platform-browser): use correct parent in animation removeChild callback

Animation's `onRemovalComplete` callback is using incorrect `parentNode`.
Should be parentNode provided by `delegate.parentNode(element)` instead of direct `element.parentNode`.
This is not a problem with default renderer but can cause problems with custom renderer if it uses another
logic to add/remove nodes.

Fixes #44023

raman-savitski commented:
> The change looks good, but the CI flaked. @raman-savitski can you try to push the same commit again so the CI can run?

Done, CI checks passed this time.

crisbeto reviewed (approved):
LGTM

AndrewKushnir commented:
[Presubmit](http://test/OCL:408629412:BASE:408629439:1636478830025:4ffb127c).

AndrewKushnir reviewed (approved):
@raman-savitski thanks for the fix 👍

atscott commented:
This PR was merged into the repository by commit 236bdff831f7dc794d5a03e2d82921887bd5eda8.
